### PR TITLE
Homework2: Fedor Polikarpov

### PR DESCRIPTION
**Pull request by FedorPolikarpov**

### CODE REVIEW

File: Задание на длинное слово.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# задание на поиск самого длинного слова
```
Comment (belovmd) [inline Задание на длинное слово.py]: please put task description

File: Задание на длинное слово.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+a = input()
+# создаем массив со знаками препинания, которые нужно удалить
+# если что, в него можно добавить необходимые знаки
+punct = ['.', ',', ':', ';', '!', '?']
```
Comment (belovmd) [inline Задание на длинное слово.py]: what about (){} ' '' etc?

File: Задание на длинное слово.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+# "разобъем" предложение в массиве и узнаем длинну массива (сколько в нем слов)
+a = a.split()
+lenn = len(a)
+f = 0
+# создаем цикл, где проверим каждое слово на наличие знаков препинания
+for word in a:
+    # если последний символ в строке знак препинания,
+    # заменим его на такое же слово но без последнего слова
```
Comment (belovmd) [inline Задание на длинное слово.py]: strip?

File: Задание на длинное слово.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    if word[-1] in punct:
+        a[f] = word[:-1]
+        word = a[f]
+    # тож самое делаем и с первым символом
```
Comment (belovmd) [inline Задание на длинное слово.py]: .... word),

File: Задание на длинное слово.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    if word[0] in punct:
+        a[f] = word[1:]
+    f += 1
+# находим самое длинное слово
+d = 0
```
Comment (belovmd) [inline Задание на длинное слово.py]: a, d, f - bad names

File: Задание на удаление повтрояющихся символов.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
+a = input()
+# удаление повторных символов
+# создадим массив с одним элементом - пробелом далее будет ясно, зачем
+words = [' ']
```
Comment (belovmd) [inline Задание на удаление повтрояющихся символов.py]: result = ''
result += symbol

File: Задание на цену.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+# Сделаем красивый вывод ответа,
+# что бы не было ( "стоит 22 рублей" или "стоит 1 копеек")
+# Для этого узнаем, на какую цифру заканчивается число рублей и копеек
+# для рублей:
+s = str(cost1)
+costrub = int(s[-1])
+# для копеек используем другой способ, без приведения к строке, но интереснее:
+costcop1, costcop2 = cost2 / 10, cost2 // 10
+costcop = int((costcop1 - costcop2) * 10)
+# делаем красивый output:
```
Comment (belovmd) [inline Задание на цену.py]: great

File: Задание на цену.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
+# Посчитать стоимость
+m = int(input('Введите стоимость вещи (руб):'))
+n = int(input('Введите стоимость вещи (коп):'))
+l = int(input('Введите колличество товара, которое хотите купить:'))
+# Узнаем, сколько руб и коп стоит вещь
+cost1 = m * l
```
Comment (belovmd) [inline Задание на цену.py]: full_cost = rub*100 + kopecks
and than just print

File: Задания на кол-во букв.py [hunk 9]
```diff
@@ -0,0 +1,21 @@
+# Задание на кол-во букв
+a = input('Введите ваше предложение: ')
+# создадим массив, в котором будут все буквы, которые мы считаем
+# сначала строчные
+massiv1 = ('qazwsxedcrfvtgbyhnujmikolp')
+small = 0
+# проверяем каждую букву в веденном предложении,
+# если такая есть в массиве, то необходимая нам переменная (b), увеличится на 1
+for i in a:
```
Comment (belovmd) [inline Задания на кол-во букв.py]: it wouldn't work with 'aabc' cause we'll have 4 small letters but your program would count only 3

File: доп. задания.py [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+# палиндром
+# на одну строчку нарушил устровие без типа str,
+# но только для того что бы решение было интереснее
+lenn = len(str(x))
+x = int(x)
+# узнал длинну числа и посделил на 2 без остатка что бы узнать,
+# сколько раз сравнивать
+iter = lenn // 2
+# добавил 2 переменные
+se, hlast = x, x
```
Comment (belovmd) [inline доп. задания.py]: I don't understand what should be kept in them

File: доп. задания.py [hunk 11]
```diff
@@ -0,0 +1,47 @@
+x = int(input('Введите число:'))
+# палиндром
+# на одну строчку нарушил устровие без типа str,
+# но только для того что бы решение было интереснее
```
Comment (belovmd) [inline доп. задания.py]: can you just build reverted number using integer operations?

**Review by belovmd — COMMENTED** [review]
1. Overall - you don't need to write in comments what are you doing. 
it's need only for tricky places and you should write why has you done in this way
2. Overall a, b, c etc - bad naming.
3. You can add additional more advanced tasks from external sites

File: Задания на кол-во букв.py (hunk 9 above)
Comment (FedorPolikarpov) [reply Задания на кол-во букв.py]: Вы не правы, она все верно считает

File: Задание на цену.py (hunk 8 above)
Comment (FedorPolikarpov) [reply Задание на цену.py]: Но тогда же full_cost будет - вся цена в копейках, я не понимаю, зачем она нам нужна

**Review by belovmd — COMMENTED** [review]
8,5

**Pull request closed without merging**